source/json.c

```diff
@@ -321,9 +313,7 @@ int aws_json_value_to_string(const struct aws_json_value *value, struct aws_byte
     }
 
     // clean the buffer before re-assigning
-    if (aws_byte_buf_is_valid(output)) {
-        aws_byte_buf_clean_up(output);
-    }
+    aws_byte_buf_clean_up(output);
```

Comment:
Ok, looking at this now, I've changed my mind on the byte_buf expectations.  The contract should be "pass in a zeroed or uninitialized byte buf only" and we should just overwrite output and not call clean up at all.  It makes no sense to ask people to init a buffer only to immediately destroy its contents and then overwrite them.

Actually I think I have a stronger preference:

```aws_byte_buf_append_json_string(struct aws_byte_buf *buffer, const struct aws_json_value *value)```

and

```aws_byte_buf_append_json_string_formatted(struct aws_byte_buf *buffer, const struct aws_json_value *value)```

which take an already-initialized byte buf and then append the json string to it.

We don't need to do any clean up and it can be used as a sub-part in something larger because we're not stomping the buffer.

For an example of something similar, where we have API functions targeting an "associated" type in the header definition of a related type see https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/string.h#L315 where we have an aws_byte_buf API in string.h